src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/events/DomainEvents.kt

```diff
@@ -6,60 +6,46 @@ import java.time.LocalDateTime
 
 interface Event
 
-data class AppointmentCreatedEvent(val additionalInformation: AppointmentInformation) :
-  DomainEvent(DomainEventType.APPOINTMENT_CREATED.eventType), Event {
-  val appointmentId = additionalInformation.appointmentId
+abstract class DomainEvent<T : AdditionalInformation>(
+  eventType: DomainEventType,
+  val additionalInformation: T,
+) {
+  val eventType = eventType.eventType
+  val description = eventType.description
+  val version: String = "1"
+  val occurredAt: LocalDateTime = LocalDateTime.now()
+
+  fun toEventType() = DomainEventType.valueOf(eventType)
 }
 
-data class AppointmentInformation(val appointmentId: Long) : AdditionalInformation
+interface AdditionalInformation
 
```

Comment:
Empty interface? What value does this provide?

Reply:
Just collecting the different event content classes into an interface, but nothing actually common between them?

Reply:
This isn't new, it was here before.  This interface is needed as it part of the message payload additionalInformation.  For each event type it would be different.  At the moment we just have the entity ID's in here but there is probably more to go in here but we just haven't got it yet e.g. base url which would be a link to the resource in question.